src/ansys/fluent/core/launcher/pyfluent_enums.py

```
@@ -104,6 +104,7 @@ class FluentMode(FluentEnum):
     SOLVER = "solver"
     SOLVER_ICING = "solver_icing"
     SOLVER_AERO = "solver_aero"
+    SOLVER_PRE_POST = "solver_pre_post"
```

@seanpearsonuk This works with only solver therefore we have given the name SOLVER_PRE_POST.

---

@hpohekar The mode is purely pre-post without solve capabilities. Hence the original choice.

---

@seanpearsonuk Yes, that's correct. I have verified it with the meshing mode as well. Thank you.

---

First option looks good to me.

---

@hpohekar OK, so we need `MESHING_PRE_POST` so that we can switch to solution and retain the pre-post limitation? Have you checked that it works?

---

@hpohekar If we change the label `SOLVER_PRE_POST` to `PRE_POST` then it actually reflects the usage *mode* rather than the _implementation_.

---

@hpohekar Have we got away without docstrings in this class up to now? This definitely should be well documented due to its importance.

---

> @hpohekar Have we got away without docstrings in this class up to now? This definitely should be well documented due to its importance.

Sure.

---

> @hpohekar OK, so we need `MESHING_PRE_POST` so that we can switch to solution and retain the pre-post limitation? Have you checked that it works? Please add test generally.

@seanpearsonuk I'm trying to find a way to test this. Please let me know if you have any idea. Thank you.

---

> @hpohekar If we change the label `SOLVER_PRE_POST` to `PRE_POST` then it actually reflects the usage _mode_ rather than the _implementation_.

@seanpearsonuk Sure, we will change it to PRE_POST. Thank you.

---

> > @hpohekar OK, so we need `MESHING_PRE_POST` so that we can switch to solution and retain the pre-post limitation? Have you checked that it works? Please add test generally.
> 
> @seanpearsonuk I'm trying to find a way to test this. Please let me know if you have any idea. Thank you.

@hpohekar 
1. Launch in `MESHING_PRE_POST` mode
2. Run a vanilla workflow and generate volume mesh
3. Switch to solver
4. Do some setup -- should be allowed
5. Try to iterate -- ideally this should cleanly raise an exception rather than hiding the error remotely (or crashing)
6. Read a dat file -- should be allowed, and previous state should not cause limiting side effects
7. Create a contour -- should be allowed

I would advise to use the steps as much as possible from an existing test that performs a switch to solver. That way we know we are on solid ground to start with.